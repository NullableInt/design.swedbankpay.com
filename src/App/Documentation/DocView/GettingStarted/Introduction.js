import React from "react";

import DocViewHeading from "../DocViewHeading";

const Introduction = () => {
    return (
        <div>
            <DocViewHeading title="Introduction" />
            <div className="doc-container">
                <div className="row">
                    <div className="col-md-12">
                        <p className="lead">Get started with the PayEx DesignGuide...</p>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Introduction;
