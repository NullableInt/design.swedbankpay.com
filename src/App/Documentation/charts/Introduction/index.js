import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc, ExperimentalComponentAlert } from "#";

const BASENAME = process.env.basename;
const scriptUrl = `https://design.payex.com${BASENAME}scripts/px-script.js`;
const dashboardScriptUrl = `https://design.payex.com${BASENAME}scripts/px.dashboard.js`;
const styleUrl = `https://design.payex.com${BASENAME}styles/px.css`;
const dashboardStyleUrl = `https://design.payex.com${BASENAME}styles/px.dashboard.css`;

const QuickStart = () => (
    <>
        <h2 id="quick-start">Quick start</h2>
        <p>To quickly add the dashboard theme to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <PrismCode className="language-html">{"<link>"}</PrismCode> into your <PrismCode className="language-html">{"<head>"}</PrismCode> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
            <link rel="stylesheet" href={dashboardStyleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>Many of our components require the use of JavaScript to function. Place the following <PrismCode className="language-html">{"<script>"}</PrismCode> tag near the end of your pages, right before the closing <PrismCode className="language-html">{"</body>"}</PrismCode> tag, to enable them.</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
            <script src={dashboardScriptUrl}></script>
        </ComponentPreview>
    </>
);

const IntroductionText = () => (
    <div className="col-lg-10 doc-body">
        <ExperimentalComponentAlert />
        <p className="lead">Get started with the charts using the PayEx DesignGuide Dashboard theme. We are using <a href="https://www.chartjs.org/" target="_blank" rel="noopener noreferrer">chart.js</a>.</p>
        <QuickStart />
        <Css />
        <JavaScript />
    </div>
);

const Introduction = () => (
    <div className="doc-container">
        <div className="row">
            <IntroductionText />
            <DocToc component={IntroductionText} />
        </div>
    </div>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript, IntroductionText };
