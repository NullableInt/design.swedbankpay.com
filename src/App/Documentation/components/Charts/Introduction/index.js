import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, Property } from "@docutils";

const BASENAME = process.env.basename;
const brand = process.env.brand;
const scriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg.js`;
const dashboardScriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg-dashboard.js`;
const styleUrl = `https://design.${brand}.com${BASENAME}styles/dg-style.css`;
const dashboardStyleUrl = `https://design.${brand}.com${BASENAME}styles/dg-dashboard.css`;

const QuickStart = () => (
    <>
        <h3 id="quick-start">Quick start</h3>
        <p>To quickly add the dashboard theme to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h3 id="css">CSS</h3>
        <p>Copy-paste the stylesheet <PrismCode className="language-html">{"<link>"}</PrismCode> into your <PrismCode className="language-html">{"<head>"}</PrismCode> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
            <link rel="stylesheet" href={dashboardStyleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h3 id="javascript">JavaScript</h3>
        <p>The dashboard theme requires a separate script to run properly. So add the dashboard script below the <Property value="scripts" /> at the bottom of your <PrismCode className="language-html">{"<body>"}</PrismCode> tag like this:</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
            <script src={dashboardScriptUrl}></script>
        </ComponentPreview>
    </>
);

const Introduction = () => (
    <>
        <h2 id="introduction">Introduction</h2>
        <p className="lead">
            We are using <a href="https://www.chartjs.org/" target="_blank" rel="noopener noreferrer">chart.js</a> to generate the charts.
            Our script is mainly just a wrapper around that library to add the correct styling etc.
        </p>
        <p>Most options and settings available in the chart.js documentation will be available.</p>
        <p>For now, the following chart types are available:</p>
        <ul>
            <li><code className="token string">{"\"line\""}</code></li>
            <li><code className="token string">{"\"bar\"/\"horizontalBar\""}</code></li>
            <li><code className="token string">{"\"pie\"/\"doughnut\""}</code></li>
        </ul>
        <QuickStart />
        {/* <Css /> */}
        <JavaScript />
    </>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript };
