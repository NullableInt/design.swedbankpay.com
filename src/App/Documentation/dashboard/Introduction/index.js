import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, Property, DocContainer } from "@docutils";

const BASENAME = process.env.basename;
const scriptUrl = `https://design.swedbankpay.com${BASENAME}scripts/px-script.js`;
const dashboardScriptUrl = `https://design.swedbankpay.com${BASENAME}scripts/px.dashboard.js`;
const styleUrl = `https://design.swedbankpay.com${BASENAME}styles/px.css`;
const dashboardStyleUrl = `https://design.swedbankpay.com${BASENAME}styles/px.dashboard.css`;

const QuickStart = () => (
    <>
        <h2 id="quick-start">Quick start</h2>
        <p>To quickly add the dashboard theme to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <PrismCode className="language-html">{"<link>"}</PrismCode> into your <PrismCode className="language-html">{"<head>"}</PrismCode> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
            <link rel="stylesheet" href={dashboardStyleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>The dashboard theme requires a separate script to run properly. So add the dashboard script below the <Property value="px-script" /> at the bottom of your <PrismCode className="language-html">{"<body>"}</PrismCode> tag like this:</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
            <script src={dashboardScriptUrl}></script>
        </ComponentPreview>
    </>
);

const Introduction = () => (
    <DocContainer>
        <p className="lead">Get started with the Swedbank Pay DesignGuide Dashboard theme.</p>
        <QuickStart />
        {/* <Css /> */}
        <JavaScript />
    </DocContainer>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript };
