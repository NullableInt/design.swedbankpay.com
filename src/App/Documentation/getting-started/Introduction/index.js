import React from "react";
import PrismCode from "react-prism";

import ComponentPreview from "../../utils/ComponentPreview";
import DocToc from "../../utils/DocToc";
import pkg from "../../../../../package.json";

const version = pkg.version;
const scriptUrl = `https://design.payex.com/${version}/scripts/px-script.js`;
const styleUrl = `https://design.payex.com/${version}/styles/px.css`;

const QuickStart = () => (
    <>
        <h2 id="quick-start">Quick start</h2>
        <p>To quickly add the designguide to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <PrismCode className="language-html">{"<link>"}</PrismCode> into your <PrismCode className="language-html">{"<head>"}</PrismCode> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure={true}>
            <link rel="stylesheet" href={styleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>Many of our components require the use of JavaScript to function. Place the following <PrismCode className="language-html">{"<script>"}</PrismCode> tag near the end of your pages, right before the closing <PrismCode className="language-html">{"</body>"}</PrismCode> tag, to enable them.</p>
        <ComponentPreview language="html" codeFigure={true}>
            <script src={scriptUrl}></script>
        </ComponentPreview>
    </>
);

const StarterTemplate = () => (
    <>
        <h2 id="starter-template">Starter template</h2>
        <p>As a starting point you can use the following template</p>
        <ComponentPreview language="html" codeFigure={true}>
            <html lang="en">
                <head>
                    <meta charSet="utf-8" />
                    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />

                    <link rel="stylesheet" href={styleUrl} />

                    <title>Hello, PayEx!</title>
                </head>
                <body>
                    <h1>Hello, PayEx!</h1>
                    <script src={scriptUrl}></script>
                </body>
            </html>
        </ComponentPreview>
    </>
);

const IntroductionText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <p className="lead">Get started with the PayEx DesignGuide.</p>
        <QuickStart />
        <Css />
        <JavaScript />
        {/* <StarterTemplate /> */}
    </div>
);

const Introduction = () => (
    <div className="doc-container">
        <div className="row">
            <IntroductionText />
            <DocToc component={IntroductionText} />
        </div>
    </div>
);

export default Introduction;

/* for testing */
export { QuickStart, Css, JavaScript, StarterTemplate, IntroductionText };
