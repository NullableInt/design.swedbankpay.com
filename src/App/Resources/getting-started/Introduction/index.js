import React from "react";

import { DocContainer } from "@docutils";

const brandTitle = process.env.brandTitle;

const Introduction = () => (
    <DocContainer docToc>
        <p className="lead">Getting started with the {brandTitle} Resources.</p>
    </DocContainer>
);

export default Introduction;
