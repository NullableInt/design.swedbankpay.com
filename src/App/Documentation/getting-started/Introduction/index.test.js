import React from "react";
import { shallow } from "enzyme";

import Introduction from "./index";

describe("GettingStarted: Introduction", () => {
    it("is defined", () => {
        expect(Introduction).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Introduction />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
