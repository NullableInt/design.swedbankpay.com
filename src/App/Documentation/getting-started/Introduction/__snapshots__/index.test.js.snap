// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted: Introduction Css renders 1`] = `
<Fragment>
  <h2
    id="css"
  >
    CSS
  </h2>
  <p>
    Copy-paste the stylesheet 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;link&gt;
    </PrismCode>
     into your 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;head&gt;
    </PrismCode>
     before all other stylesheets to load our CSS.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <link
      href="https://design.payex.com/TESTstyles/px.css"
      rel="stylesheet"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`GettingStarted: Introduction IntroductionText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    Get started with the PayEx DesignGuide.
  </p>
  <QuickStart />
  <Css />
  <JavaScript />
</div>
`;

exports[`GettingStarted: Introduction JavaScript renders 1`] = `
<Fragment>
  <h2
    id="javascript"
  >
    JavaScript
  </h2>
  <p>
    Many of our components require the use of JavaScript to function. Place the following 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;script&gt;
    </PrismCode>
     tag near the end of your pages, right before the closing 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;/body&gt;
    </PrismCode>
     tag, to enable them.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <script
      src="https://design.payex.com/TESTscripts/px-script.js"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`GettingStarted: Introduction QuickStart renders 1`] = `
<Fragment>
  <h2
    id="quick-start"
  >
    Quick start
  </h2>
  <p>
    To quickly add the designguide to your project, include the snippets below in your project.
  </p>
</Fragment>
`;

exports[`GettingStarted: Introduction renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <IntroductionText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
